homeassistant/components/minecraft_server/api.py

```diff
@@ -52,35 +54,54 @@ class MinecraftServerConnectionError(Exception):
     """Raised when no data can be fechted from the server."""
 
 
+class MinecraftServerNotInitializedError(Exception):
+    """Raised when APIs are used although server instance is not initialized yet."""
+
+
 class MinecraftServer:
     """Minecraft Server wrapper class for 3rd party library mcstatus."""
 
     _server: BedrockServer | JavaServer
 
-    def __init__(self, server_type: MinecraftServerType, address: str) -> None:
+    def __init__(
+        self, hass: HomeAssistant, server_type: MinecraftServerType, address: str
+    ) -> None:
         """Initialize server instance."""
+        self._initialized = False
```

Comment:
`_server` is no longer initialized in the constructor, and so i think it also needs to support the `None` type. I would suggest dropping `_initialized` and using a check for `_server is not None` as detecting of things were initialized.

Reply:
Sounds like a good idea 👍